spec/use_case/address_list_spec.rb

```diff
@@ -0,0 +1,20 @@
+describe PostcodeValidation::UseCase::AddressList do
+  let(:address_list_gateway) do
+    double(query: address_list_response)
+  end
+  subject do
+    described_class.new(address_list_gateway: address_list_gateway)
+      .execute(postcode: postcode, country: country)
+  end
+
+  context 'given a valid postcode and country' do
+    let(:address_list_response) { [{'StreetAddress': 'Southwark Street', 'Place': '136'}] }
+    let(:postcode) { 'SE10SW' }
+    let(:country) { 'GB' }
+    it 'returns a list of addresses' do
+      expected_result = [PostcodeValidation::Domain::AddressList.new('StreetAddress' => 'Southwark Street', 'Place' => '136')]
```

Comment:
This only tests the happy path, as anything else would be duplicating the other usecase test.
(error states)